jax/_src/numpy/lax_numpy.py

```diff
@@ -1368,6 +1368,87 @@ def gradient(f, *varargs, axis: Optional[Union[int, Tuple[int, ...]]] = None,
 def isrealobj(x):
   return not iscomplexobj(x)
 
+@partial(jit, static_argnums=1)
+class _RankWarning(UserWarning):
+    """
+    Issued by `polyfit` when the Vandermonde matrix is rank deficient.
+    For more information, a way to suppress the warning, and an example of
+    `RankWarning` being issued, see `polyfit`.
+    """
+    pass
+
+@_wraps(np.polyfit)
+def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
+  _check_arraylike("polyfit", x, y, deg)
+  x = core.concrete_or_error(None, x, "The error arose in polyfit()")
+  y = core.concrete_or_error(None, y, "The error arose in polyfit()")
```

Comment:
I don't think there's any need to ensure `x` and `y` are concrete. Looking through the code below, it seems like they could be traced quantites, except perhaps for the `rank` check (which I would consider removing in order to allow JIT compilation)